### PR TITLE
fix topic subscribing bug

### LINKED ISSUE #425: Event subscribing bug
With regard to event subscribe, topics that is not subscribed are also delivered from the server.
For example, `/ timer / sec / 10` and` / timer / sec / 100` are also subscribed if I subscribe `/ timer / sec / 1`.
This is because go-radix does a string-level search that does not treat `/` specially.

### PR DESCRIPTION
**Pull request by YujiOshima**
fix #425 
Signed-off-by: Yuji Oshima <yuji.oshima0x3fd@gmail.com>

### CODE REVIEW

File: pkg/broker/server/sse.go [hunk 1]
```diff
@@ -244,6 +245,15 @@ func (b *Broker) run() {
...
 					}
+					if key != "/" {
+						// Make sure that the topic subscribed to by the client is the upper topic of the topic being notified or the topic exactly matched.
+						notfyTopic := strings.Split(event.topic, "/")
+						for i, sliceTopic := range strings.Split(key, "/") {
+							if notfyTopic[i] != sliceTopic {
+								return false
+							}
+						}
+					}
```
Comment (chungers) [inline pkg/broker/server/sse.go]: Clever to do the check here.

Would you mind put this code in a function?   Something like  `if !checkPath(key, event.topic) { return false }`?

Comment (chungers) [inline pkg/broker/server/sse.go]: s/notfyTopic/notifyTopic/g

**Comment by chungers** [comment]
@YujiOshima  Thank you for catching and fixing this bug and adding tests.

Once you move the check to a function I will merge it in.

I still think a better way is to intercept the subscribe call and reject a topic that doesn't conform to the spec of the Event plugin.  This way, we don't have to do this check at every outbound event publish.  It's also better UX since the client knows it tried to subscribe to an incorrect topic.

One place this might work, would be to add checks as an interceptor to the ServerHTTP handler.  This would be the place: https://github.com/docker/infrakit/blob/master/pkg/rpc/server/server.go#L132

Here we have references to all the `event.Plugin` implementations and we can wrap the `broker.ServeHTTP` handler with a handler that does proper topic validation before they are forwarded on to the real thing.  This would also give meaningful HTTP error code on subscription, rather than events quietly dropped to the floor.   Finally the `pkg/broker` doesn't have to have any knowledge of `/` convention for topics and can be used for something that uses a different topic naming convention (maybe `.object1.property1`).

File: pkg/broker/server/sse.go (hunk 1 above)
Comment (YujiOshima) [inline pkg/broker/server/sse.go]: Thank you, moved to function!

**Pull request merged**